docs/migrations/v0.40.md

```diff
@@ -0,0 +1,381 @@
+changing from the manifest containing a list of kinds, each of which contain one or more versions, 
+to the manifest having a set of versions, each of which contain one or more kinds (which are version-specific).
+
+In practical terms, this means that a simple manifest like:
+```cue
+manifest: {
+    appName: "example"
+    kinds: [{
+        kind: "Foo"
+        versions: {
+            "v1": {
+                schema: {
+                    spec: {
+                        field: string
+                    }
+                }
+            }
+            "v2": {
+                schema: {
+                    spec: {
+                        breakingChange: string
+                    }
+                }
+            }
+        }
+    }]
+}
+```
+would now look like:
+```cue
+manifest: {
+    appName: "example"
+    versions: {
+        "v1": {
+            kinds: [{
+                kind: "Foo"
+                schema: {
+                    spec: {
+                        field: string
+                    }
+                }
+            }]
+        }
+        "v2": {
+            kinds: [{
+                kind: "Foo"
+                schema: {
+                    spec: {
+                        breakingChange: string
+                    }
+                }
+            }]
+        }
+    }
+}
+```
+
+For a more complex example, where kinds are defined separately from the manifest, there is a pattern which can be adopted to avoid duplicating data for each version of the kind:
+```cue
+manifest: {
+    appName: "example"
+    kinds: [foo, bar]
+}
+
+foo: {
+    kind: "Foo"
+    plural: "multifoo"
+    conversion: true
+    validation: operations: ["CREATE","UPDATE"]
+    versions: {
+        "v1": {
+            schema: {
+                spec: {
+                    foo: string
+                }
+            }
+        }
+        "v2": {
+            schema: {
+                spec: {
+                    bar: string
+                }
+            }
+        }
+    }
+}
+
+bar: {
+    kind: "Bar"
+    versions: {
+        "v1": {
+            schema: {
+                spec: {
+                    foz: int
+                }
+            }
+        }
+        "v2": {
+            validation: operations: ["CREATE","UPDATE]
+            schema: {
+                spec: {
+                    baz: int
+                }
+            }
+        }
+    }
+}
+```
+would be converted to:
+```cue
+manifest: {
+    appName: "example"
+    versions: {
+        "v1": {
+            kinds: [foov1, barv1]
+        }
+        "v2": {
+            kinds: [foov2, barv2]
+        }
+    }
+}
+
+fooKind: {
+    kind: "Foo"
+    plural: "multifoo"
+    conversion: true
+    validation: operations: ["CREATE","UPDATE"]
+}
+
+foov1: fooKind & { // join the kind data into this new object which contains the schema for the version
+    schema: {
+        spec: {
+            foo: string
+        }
+    }
+}
```

Comment:
Maybe its only for the demo what you did or I miss context? If I get this right, those fields should be emitted from the manifest generation or they are skipped even if they appear? 

I would expect something like the following 🤔 
```
#fooKind: {
	kind:       "Foo"
	plural:     "multifoo"
	conversion: true
	validation: operations: ["CREATE", "UPDATE"]
    ...
}

#foov1: #fooKind & {
	schema: {
		spec: {
			foo: string
		}
	}
}
```

Reply:
The codegen will only pull the `manifest` key from your CUE (unless you change it with a CLI flag), so you don't need to make the other fields definitions.